tools/building.py

```diff
@@ -458,9 +462,13 @@ def link_lld(args, target, external_symbols=None):
   # wasm-ld needs to take this for the LTO mode.
   if settings.EXCEPTION_HANDLING:
     cmd += ['-mllvm', '-exception-model=wasm', '-mllvm', '-wasm-enable-eh']
+  # Wasm SjLj handling. This is a CodeGen option for the LLVM backend, so
+  # wasm-ld needs to take this for the LTO mode.
+  if settings.EXCEPTION_HANDLING or settings.SJLJ_HANDLING:
+    cmd += ['-mllvm', '-exception-model=wasm']
```

Comment:
It's nice that lld has a single flag for both wasm exceptions and wasm sjlj. I wonder if we can do the same in emscripten somehow, and avoid adding a new flag here? But I guess we can't because we want the possibility to use wasm sjlj without C++ exceptions, since if we enable exceptions we change codegen and which system libraries are used. I'm curious how lld avoids that - I guess since it is not in charge of C++ semantics, and doesn't pick system libraries?

Reply:
I see, thanks!